security.php

```diff
@@ -54,56 +55,148 @@ function wpcom_vip_is_restricted_username( $username ) {
+	 * @param string $ip        IP address of the login request.
+	 * @param string $username  Username of the login request.
+	 */
+	$ip_username_threshold = apply_filters( 'wpcom_vip_ip_username_login_threshold', 3, $ip, $username );
+
+	/**
+	 * Filters the threshold for limiting logins by IP.
+	 *
+	 * @param int    $threshold IP login threshold.
+	 * @param string $ip        IP address of the login request.
+	 */
+	$ip_threshold = apply_filters( 'wpcom_vip_ip_login_threshold', 5, $ip );
+
+	/**
+	 * Filters the threshold for limiting logins by username.
+	 *
+	 * @param int    $threshold Username login threshold.
+	 * @param string $username  Username of the login request.
+	 */
+	$username_threshold = apply_filters( 'wpcom_vip_username_login_threshold', 10, $username );
+
+	/**
+	 * Change the thresholds for Password Resets
+	 */
+	if ( 'lost_password_limit' === $cache_group ) {
+		/**
+		 * Filters the threshold for limiting password resets by IP + username combination.
+		 *
+		 * Note that changing this value will also change the default value for the username password reset threshold as well,
+		 * which is set as 5 times this value.
+		 *
+		 * @param int    $threshold IP + Username combination password reset threshold. Default 3.
+		 * @param string $ip        IP address of the password reset request.
+		 * @param string $username  Username of the password reset request.
+		 */
+		$ip_username_threshold = apply_filters( 'wpcom_vip_ip_username_password_reset_threshold', 3, $ip, $username );
+
+		/**
+		 * Filters the threshold for limiting password resets by IP.
+		 *
+		 * @param int    $threshold IP password reset threshold. Default 3.
+		 * @param string $ip        IP address of the password reset request.
+		 */
+		$ip_threshold = apply_filters( 'wpcom_vip_ip_password_reset_threshold', 3, $ip );
+
+		/**
+		 * Filters the threshold for limiting password resets by username.
+		 *
+		 * @param int    $threshold Username password reset threshold. Default is 5 times the IP + username combination threshold.
+		 * @param string $username  Username of the password reset request
+		 */
+		$username_threshold = apply_filters( 'wpcom_vip_username_password_reset_threshold', 5 * $ip_username_threshold, $username );
+	} elseif ( $is_restricted_username ) {
+		$ip_username_threshold = 2;
+	}
+
+	if ( $ip_username_count >= $ip_username_threshold || $ip_count >= $ip_threshold || $username_count >= $username_threshold ) {
+		$lock_interval = apply_filters( 'wpcom_vip_invalid_login_lock_period', MINUTE_IN_SECONDS * 30  );
+		if ( $cache_group === CACHE_GROUP_LOGIN_LIMIT ) {
+			do_action( 'login_limit_exceeded', $username );
+		} else {
+			do_action( 'password_reset_limit_exceeded', $username );
+		}
+
+		if ( $ip_username_count >= $ip_username_threshold ) {
+			wp_cache_set( CACHE_KEY_LOCK_PREFIX . $cache_keys['ip_username_cache_key'], true, $cache_group, $lock_interval );
+		} else if ( $ip_count >= $ip_threshold ) {
```

Comment:
```suggestion
		} elseif ( $ip_count >= $ip_threshold ) {
```

SonarCloud insists on this 🤷🏽‍♂️

Reply:
It's also required per WPCS as well.